scripts/pylib/twister/twisterlib.py

```diff
@@ -918,34 +953,8 @@ def handle(self):
                              args=(ser, read_pipe, harness))
         t.start()
 
-        d_log = "{}/device.log".format(self.instance.build_dir)
-        logger.debug('Flash command: %s', command)
-        try:
-            stdout = stderr = None
-            with subprocess.Popen(command, stderr=subprocess.PIPE, stdout=subprocess.PIPE) as proc:
-                try:
-                    (stdout, stderr) = proc.communicate(timeout=60)
-                    # ignore unencodable unicode chars
-                    logger.debug(stdout.decode(errors = "ignore"))
-
-                    if proc.returncode != 0:
-                        self.instance.status = "error"
-                        self.instance.reason = "Device issue (Flash error?)"
-                        with open(d_log, "w") as dlog_fp:
-                            dlog_fp.write(stderr.decode())
-                        os.write(write_pipe, b'x')  # halt the thread
-                except subprocess.TimeoutExpired:
-                    logger.warning("Flash operation timed out.")
-                    proc.kill()
-                    (stdout, stderr) = proc.communicate()
-                    self.instance.status = "error"
-                    self.instance.reason = "Device issue (Timeout)"
-
-            with open(d_log, "w") as dlog_fp:
-                dlog_fp.write(stderr.decode())
-
-        except subprocess.CalledProcessError:
-            os.write(write_pipe, b'x')  # halt the thread
+        if not flash_before:
+            self.run_flash(command)
```

Comment:
so esp does not support `west flash`, and can not use the west flash command?

Reply:
I did not get the point in here. It does support west flash.

Reply:
sorry my mis-understanding your code.  it is not related